droidlet/lowlevel/minecraft/pyworld/physical_interfaces.py

```diff
@@ -0,0 +1,231 @@
+            self.world.step()
+
+
+class FakeCPPAction:
+    NAME = "NULL"
+
+    def __init__(self, agent):
+        self.agent = agent
+
+    def action(self, *args):
+        self.agent.world_interaction_occurred = True
+
+    def __call__(self, *args):
+        if hasattr(self.agent, "recorder"):
+            self.agent.recorder.record_action({"name": self.NAME, "args": list(args)})
+        return self.action(*args)
+
+
+class Dig(FakeCPPAction):
+    NAME = "dig"
+
+    def action(self, x, y, z):
+        self.agent.world_interaction_occurred = True
+        dug = self.agent.world.dig((x, y, z))
+        if dug:
+            self.agent._changed_blocks.append(((x, y, z), (0, 0)))
+            return True
+        else:
+            return False
+
+
+class SendChat(FakeCPPAction):
+    NAME = "send_chat"
+
+    def action(self, chat):
+        self.agent.world_interaction_occurred = True
+        logging.info("FakeAgent.send_chat: {}".format(chat))
+        self.agent._outgoing_chats.append(chat)
+
+
+class SetHeldItem(FakeCPPAction):
+    NAME = "set_held_item"
+
+    def action(self, arg):
+        self.agent.world_interaction_occurred = True
+        try:
+            d, m = arg
+            self.agent._held_item = (d, m)
+        except TypeError:
+            self.agent._held_item = (arg, 0)
+
+
+class StepPosX(FakeCPPAction):
+    NAME = "step_pos_x"
+
+    def action(self):
+        self.agent.world_interaction_occurred = True
+        self.agent.pos += (1, 0, 0)
+
+
+class StepNegX(FakeCPPAction):
+    NAME = "step_neg_x"
+
+    def action(self):
+        self.agent.world_interaction_occurred = True
+        self.agent.pos += (-1, 0, 0)
+
+
+class StepPosZ(FakeCPPAction):
+    NAME = "step_pos_z"
+
+    def action(self):
+        self.agent.world_interaction_occurred = True
+        self.agent.pos += (0, 0, 1)
+
+
+class StepNegZ(FakeCPPAction):
+    NAME = "step_neg_z"
+
+    def action(self):
+        self.agent.world_interaction_occurred = True
+        self.agent.pos += (0, 0, -1)
+
+
+class StepPosY(FakeCPPAction):
+    NAME = "step_pos_y"
+
+    def action(self):
+        self.agent.world_interaction_occurred = True
+        self.agent.pos += (0, 1, 0)
+
+
+class StepNegY(FakeCPPAction):
+    NAME = "step_neg_y"
+
+    def action(self):
+        self.agent.world_interaction_occurred = True
+        self.agent.pos += (0, -1, 0)
+
+
+class StepForward(FakeCPPAction):
+    NAME = "step_forward"
+
+    def action(self):
+        self.agent.world_interaction_occurred = True
+        dx, dy, dz = self.agent._look_vec
+        self.agent.pos += (dx, 0, dz)
+
+
+class TurnAngle(FakeCPPAction):
+    NAME = "turn_angle"
+
+    def action(self, angle):
+        self.agent.world_interaction_occurred = True
+        if angle == 90:
+            self.agent.turn_left()
+        elif angle == -90:
+            self.agent.turn_right()
+        else:
+            raise ValueError("bad angle={}".format(angle))
+
+
+# FIXME!
```

Comment:
not sure I understand what needs to be fixed here ?

Reply:
its an old fixme, just copied over.   but we aren't using turn_left and turn_right anywhere, I think these are dead now; the fixme was to investigate and torch